src/main/java/com/github/mc1arke/sonarqube/plugin/server/CommunityComponentKey.java

```diff
@@ -29,13 +30,15 @@
 
     private final String key;
     private final String dbKey;
-    private final BranchSupport.Branch branch;
+    private final String branchName;
+    private final BranchType branchType;
     private final String pullRequestKey;
 
-    /*package*/ CommunityComponentKey(String key, String dbKey, BranchSupport.Branch branch, String pullRequestKey) {
+    /*package*/ CommunityComponentKey(String key, String dbKey, String branchName, BranchType branchType, String pullRequestKey) {
```

Comment:
The `BranchType` is probably unnecessary here. `ReportSubmitter` in Sonarqube core will only call the `Community ranchSupportDelegate#createBranchComponent` method if the branch does not already exist, and will be passing in the 'main' branch as the component DTO for comparison, and for basing any new component on. The branch type would only therefore make a difference if the new component matched the name of the main branch but was a pull request, but this is specifically defended against in `ReportSubmitter`

Reply:
There was a specific condition at `CommunityBranchSuppirtDelegate:93` that I maintained by passing the `branchType` arg. Do you think that condition is not necessary anymore because of some changes in 8.5 or it already was unnecessary? If it is so, we can simplify a lot.